test/test.js

```diff
@@ -1013,4 +1013,35 @@ describe('Overall test', () => {
         done();
     });
 
+    it ('mysql.query should use the same key when retrying', (done) => {
+        const mysql = new CustomMySQL();
+        const key = 'key';
+        const key2 = 'key2';
+
+        mysql.set_logger(noop_logger)
+            .add(key, FREE_DB)
+            .add(key2, FREE_DB2)
+            .use(key)
+            .query('CREATE TABLE users(id int NOT NULL);', (err, result) => {
+
+                should.equal(err, null);
+                result.should.exist;
+
+                mysql.use(key)
+                    .retry_if(['ER_BAD_NULL_ERROR'])
+                    .build('INSERT INTO users (SELECT IF((@tmp := COALESCE(@tmp, 0) + COALESCE(@tmp, 1))=1, NULL, @tmp))')
+                    .promise()
+                    .then(() => done())
```

Comment:
Shouldn't this log an error since we are not expecting this to work?

Reply:
it will work on the 2nd try

Reply:
What do you mean by the second try? This shouldn't be called at all right? (the retry should take care of the status until it comes to a conclusion?)

Reply:
I think he means that it will fail first with 'ER_BAD_NULL_ERROR' since the query
`SELECT IF((@tmp := COALESCE(@tmp, 0) + COALESCE(@tmp, 1))=1, NULL, @tmp)`
will result to NULL at first and setting `@tmp` to `1`. The next run of the query will then result to `2`
which will return success.

Reply:
Then this is a positive test case, which shouldn't have the cleanup in the catch code.

(Also the catch code doesn't call done so this test is never finished?)